gk/main.c

```diff
@@ -410,9 +434,8 @@ gk_process_granted(struct flow_entry *fe, struct ipacket *packet,
 		priority = PRIORITY_RENEW_CAP;
 	}
 
-	/*
-	 * TODO If the nexthop MAC address is stale, then drop the packet.
-	 */
+	eth_cache = fib->u.grantor.eth_cache;
+	RTE_VERIFY(eth_cache != NULL);
```

Comment:
Move these two lines to immediatly before `if (pkt_copy_cached_eth_header(packet->pkt, eth_cache))`; this improves readability.